doc/api/module.md

```diff
@@ -113,7 +113,9 @@ changes:
     `initialize` hook.
 
 Register a module that exports [hooks][] that customize Node.js module
-resolution and loading behavior. See [Customization hooks][].
+resolution and loading behavior. Registering modules that export [hooks][] only
+has an effect on the main thread. All worker threads inherit the customization
+hooks chain of the main thread. See [Customization hooks][].
```

Comment:
```suggestion
resolution and loading behavior. These hooks are run in a new thread
spawned for this purpose. After registration, the hooks will apply to any
modules imported on the main thread or on any worker threads created
by application code. See [Customization hooks][].
```

I know that for the main thread, obviously the hooks don’t apply to any modules already imported before the hooks are registered, and then they do apply to modules imported after registration; is it the same for worker threads? So if I have code that creates a new worker thread, then registers hooks, and then code within the worker thread does `import()`, will that last `import()` be customized?

And what happens when you call `register` from a worker thread? If it just silently does nothing, we should probably change that to have it throw an error. I think it should either work as intended (adding new hooks to the customization chains) from wherever it’s called, or it should error when called from places where it doesn’t do its job.